vshard.go

```diff
@@ -92,13 +92,21 @@ type Config struct {
 	TopologyProvider TopologyProvider // TopologyProvider is required provider
 
 	// Discovery
-	DiscoveryTimeout time.Duration // DiscoveryTimeout is timeout between cron discovery job; by default there is no timeout
+	// DiscoveryTimeout is timeout between cron discovery job; by default there is no timeout.
+	DiscoveryTimeout time.Duration
 	DiscoveryMode    DiscoveryMode
 
 	TotalBucketCount uint64
 	User             string
 	Password         string
 	PoolOpts         tarantool.Opts
+
+	// BucketGetter is an optional argument.
+	// You can specify a function that will receive the bucket id from the context.
+	// This is useful if you use middleware that inserts the calculated bucket id into the request context.
+	BucketGetter func(ctx context.Context) uint64
+	// RequestTimeout timeout for requests to Tarantool.
+	RequestTimeout time.Duration
```

Comment:
How we set request timeout in go-tarantool library? Can we use the same approach? Why we need another one RequestTimeout config here? I guess this may cause misleadings and confusion, because we already have request Timeout parameter in PoolOpts tarantool.Opts above. We should try to keep interfaces as simple as possible.

Reply:
yes, timeouts are now our sore point, let's leave it for now, but in the future we will plan a global rework

Reply:
Yes, now this is happening badly, and timeouts work now are extremely unclear, but this is the only more or less convenient way to use sugar

Reply:
I added a more detailed description

Reply:
We discussed this config in offline, so we have decided to remove `RequestTimeout` and use `r.cfg.PoolOpts.Timeout` instead.

Reply:
that the pool.Opts timeout is a timeout of the go-tarantool client, in our case it is essentially a timeout of the library itself
This is the difference between the timeout of the library itself
that is, our retry timeout if the buckets, for example, move

Reply:
Yes, it makes sense to me now.
Could you add this comment here in the code to avoid the same question in the future?